java/arcs/core/util/UnionFind.kt

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.core.util
+
+/**
+ * A union-find datastructure for computing equivalence classes consisting of [E] instances.
+ * This datastructure is not thread safe.
+ */
+class UnionFind<E> {
+    /**
+     * A node in the union find datastructure.
+     */
+    private data class Node<E>(val element: E, var parent: Node<E>? = null)
```

Comment:
Child classes should go at the bottom of the parent class.

Reply:
Oh, yeah! I thought I had moved it. Thanks for catching it.